### DISCUSSION TITLE
Python console hints

### DISCUSSION THREAD

**Discussion by uniquadev** [Vector35/binaryninja-api#5433] [Ideas] [answered]
I've been enjoying writing my first C++ plugin with Binary Ninja recently. To streamline my workflow, I often refer to the Python API and test features before implementing them in C++. Using VSCode, I can quickly look up methods and properties with the Python language server before moving to Binary Ninja and trying them in the console. It would be incredibly useful to have autocomplete hints directly in the Binary Ninja console. This feature would enhance the workflow and encourage more efficient and straightforward operations.

**Comment by plafosse** [comment, marked as answer]
What other than the existing tab completion would you like?
<img width="887" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/10556585/c60ba664-8b50-4bd8-a3b4-32e4d3fba9b4">

**Reply by uniquadev** [reply]
Amazing, if possible a syntax highlighter will make the console even better!

**Reply by psifertex** [reply]
To be clear, that tab-completion is already on the existing console and has been for some time, it's not new. I don't think a syntax highlighter makes sense for something that is only intended for a single line or a short selection of multi-line. For that, I would recommend using the Snippets plugin which does include a syntax highlighter and is meant for larger scripts. Or, you can even edit snippets directly in VSCode to get a fully-featured IDE. No way we'll compete with a true IDE long-term.

**Reply by uniquadev** [reply]
I apologize for the oversight. I didn't expect that pressing 'Tab' would make the completion tab open. I assumed it would appear while typing.

**Reply by psifertex** [reply]
No worries, that's useful feedback even just knowing it wasn't intuitive. We'll have to think about ways we might be able to make it more obvious.
